crates/autopilot/src/domain/competition/winner_selection/combinatorial.rs

```diff
@@ -350,3 +350,748 @@ type ScoreByDirection = HashMap<DirectedTokenPair, Score>;
+                let solution_uid = solution_map.get(&solution_id).unwrap().solution().id;
+                assert!(solutions.iter().any(|s| s.solution().id == solution_uid));
+            }
+
+            // select the winners
+            let solutions = arbitrator.mark_winners(solutions);
+            let winners = filter_winners(&solutions);
+            assert_eq!(winners.len(), self.expected_winners.len());
+            for solution_id in self.expected_winners.clone() {
+                let solution_uid = solution_map.get(&solution_id).unwrap().solution().id;
+                assert!(winners.iter().any(|s| s.solution().id == solution_uid));
+            }
+
+            // compute reference score
+            let reference_scores = arbitrator.compute_reference_scores(&solutions);
+            for (solver_id, expected_score) in self.expected_reference_scores.clone() {
+                let solver_address: eth::Address = (*solver_map.get(&solver_id).unwrap()).into();
+                let score = reference_scores.get(&solver_address).unwrap();
+                assert_eq!(score.0, eth::Ether(expected_score))
+            }
+        }
+    }
+
+    #[serde_as]
+    #[derive(Deserialize, Debug)]
+    pub struct TestAuction {
+        pub orders: HashMap<String, TestOrder>,
+        #[serde(default)]
+        #[serde_as(as = "Option<HashMap<_, U256FromDecimalStr>>")]
+        pub prices: Option<HashMap<String, eth::U256>>,
+    }
+
+    #[serde_as]
+    #[derive(Deserialize, Debug, Clone)]
+    pub struct TestOrder(
+        // sell_token
+        pub String,
+        // sell_amount
+        #[serde_as(as = "U256FromDecimalStr")] pub eth::U256,
+        // buy_token
+        pub String,
+        // buy_amount
+        #[serde_as(as = "U256FromDecimalStr")] pub eth::U256,
+    );
+
+    #[derive(Deserialize, Debug)]
+    pub struct TestSolution {
+        pub solver: String,
+        pub trades: HashMap<String, TestTrade>,
+        pub score: eth::U256,
+    }
+
+    #[serde_as]
+    #[derive(Deserialize, Debug)]
+    pub struct TestTrade(
+        // sell_amount
+        #[serde_as(as = "U256FromDecimalStr")] pub eth::U256,
+        // buy_amount
+        #[serde_as(as = "U256FromDecimalStr")] pub eth::U256,
+    );
+
+    fn create_test_arbitrator() -> super::Config {
+        super::Config {
+            max_winners: 10,
+            weth: H160::from_slice(&hex!("C02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2")).into(),
+        }
+    }
+
+    fn address(id: u64) -> H160 {
+        H160::from_low_u64_le(id)
+    }
+
+    fn create_order(
+        uid: u64,
+        sell_token: H160,
+        sell_amount: eth::U256,
+        buy_token: H160,
+        buy_amount: eth::U256,
+    ) -> Order {
+        let mock_protocol_fees = vec![fee::Policy::Surplus {
+            factor: fee::FeeFactor::try_from(0.0).unwrap(),
+            max_volume_factor: fee::FeeFactor::try_from(0.0).unwrap(),
+        }];
```

Comment:
Couldn't this just be an empty vector instead? This fee policy shouldn't capture any fees and therefor not change the outcome unless I'm missing something.